crates/sui-types/src/transaction.rs

```diff
@@ -2378,6 +2380,11 @@ impl SenderSignedData {
                 error: "zklogin is not enabled on this network".to_string(),
             });
         }
+        if !config.passkey_auth() && self.has_passkey_sig() {
```

Comment:
btw I have refactored this function, you may have to rebase. The checks are now done with the matches